spotbugs_ignore.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<FindBugsFilter>
+  <Match>
+    <!-- Uses tryWithResources, thus closes resource properly:
+    [ERROR] Medium: org.apache.rat.configuration.builders.ChildContainerBuilder.setResource(String) may fail to close stream
+    [org.apache.rat.configuration.builders.ChildContainerBuilder] At ChildContainerBuilder.java:[line 62] OS_OPEN_STREAM
+    <Class name="org.apache.rat.configuration.builders.ChildContainerBuilder"/>
+    <Bug pattern="OS_OPEN_STREAM"/-->
+  </Match>
```

Comment:
If the file is missing the build fails, therefore I had to introduce it in all submodules.

Reply:
The maven spotbugs integration allows either to run spotbugs:spotbugs to see the results and prints its results into the build log. Apart from that it is integrated into the site build so that each submodule has a spotbugs-report.

Spotbugs allows to configure a threshold ... I'll see if I can integrate this to at least fail the builds once a new error is introduced:
https://spotbugs.github.io/spotbugs-maven-plugin/spotbugs-mojo.html

Reply:
Unfortunately this is not what we would want to have :( The docs are not really that good as no description is provided.

Reply:
The only available exclusion rule seems to relate to the severity ....
https://github.com/spotbugs/spotbugs-maven-plugin/blob/master/src/main/groovy/org/codehaus/mojo/spotbugs/BaseViolationCheckMojo.groovy

Reply:
I assume mvn spotbugs:gui is the best way to work with the tool at the moment as we generate the webpage manually before an upcoming release .....